build-system/common/utils.js

```diff
@@ -96,21 +96,24 @@ function logFiles(files) {
 }
 
 /**
- * Extracts the list of files from argv.files.
+ * Extracts the list of files from argv.files. Throws an error if no matching
+ * files were found.
  *
  * @return {Array<string>}
  */
 function getFilesFromArgv() {
-  // TODO: https://github.com/ampproject/amphtml/issues/30223
-  // Switch from globby to a lib that supports Windows.
+  if (!argv.files) {
+    return [];
+  }
+  // TODO(#30223): globby only takes posix globs. Find a Windows alternative.
   const toPosix = (str) => str.replace(/\\\\?/g, '/');
-  return argv.files
-    ? globby.sync(
-        (Array.isArray(argv.files) ? argv.files : argv.files.split(','))
-          .map((s) => s.trim())
-          .map(toPosix)
-      )
-    : [];
+  const globs = Array.isArray(argv.files) ? argv.files : argv.files.split(',');
+  const files = globby.sync(globs.map((glob) => glob.trim()).map(toPosix));
+  if (files.length == 0) {
+    log(red('ERROR:'), 'Argument(s)', cyan(argv.files), 'matched zero files.');
+    throw new Error('No tests to run.');
```

Comment:
```suggestion
    throw new Error('No test files found matching specified file pattern.');
```

Reply:
This info is already contained in the error message on line 113.

Reply:
Many places in `build-system` which explicitly `log` an error message then use the same string (or a replicated version without colors) as the message for the error thrown. I think that makes more sense here. Saying "no tests to run" has a different implication.

Examples:
https://github.com/ampproject/amphtml/blob/main/build-system/tasks/runtime-test/runtime-test-base.js#L402-L404
https://github.com/ampproject/amphtml/blob/main/build-system/tasks/check-sourcemaps.js#L58-L157

Reply:
Fair enough. I've used the same message without the (potentially long) list of args.